toolkit/tools/internal/pkgjson/pkgjson.go

```diff
@@ -339,78 +338,56 @@ func (interval *PackageVerInterval) Compare(other *PackageVerInterval) (result i
 	return equalTo
 }
 
-// versionInInterval returns true if ver is a version in the set of versions represented in interval
-func (interval *PackageVerInterval) versionInInterval(ver *versioncompare.TolerantVersion) (valid bool) {
-	if ver.Compare(interval.LowerBound) == 0 && interval.LowerInclusive {
-		return true
-	}
-	if ver.Compare(interval.UpperBound) == 0 && interval.UpperInclusive {
-		return true
-	}
-	if ver.Compare(interval.UpperBound) < 0 && ver.Compare(interval.LowerBound) > 0 {
-		return true
+// Contains checks if the interval fully contains the query range
+func (interval *PackageVerInterval) contains(queryInterval *PackageVerInterval) (contains bool) {
+	lowerBoundValid := false
+	upperBoundValid := false
+
+	// Check if the each bound of the query is within the interval, or if it is the same as the bound of the interval
+	// and the bound is inclusive.
+	if interval.LowerBound.LT(queryInterval.LowerBound) ||
+		(interval.LowerBound.EQ(queryInterval.LowerBound) && interval.LowerInclusive) ||
+		(interval.LowerBound.EQ(queryInterval.LowerBound) && (interval.LowerInclusive == queryInterval.LowerInclusive)) {
+		lowerBoundValid = true
 	}
-	return false
-}
 
-// Contains checks if the interval fully contains the query range
-func (interval *PackageVerInterval) Contains(queryInterval *PackageVerInterval) (contains bool) {
-	// Check the lower bound
-	lowerBoundValid := interval.versionInInterval(queryInterval.LowerBound)
-	upperBoundValid := interval.versionInInterval(queryInterval.UpperBound)
+	if interval.UpperBound.GT(queryInterval.UpperBound) ||
+		(interval.UpperBound.EQ(queryInterval.UpperBound) && interval.UpperInclusive) ||
+		(interval.UpperBound.EQ(queryInterval.UpperBound) && (interval.UpperInclusive == queryInterval.UpperInclusive)) {
+		upperBoundValid = true
+	}
 
 	return lowerBoundValid && upperBoundValid
 }
 
-// Satisfies returns true the query interval overlaps the current interval at any point (ie it provides a version of a package
-// which satisfied the query)
-func (interval *PackageVerInterval) Satisfies(queryInterval *PackageVerInterval) (valid bool) {
-	var (
-		queryUpperValid, queryLowerValid, superset bool
-	)
-	if interval.LowerBound.Compare(queryInterval.UpperBound) > 0 || interval.UpperBound.Compare(queryInterval.LowerBound) < 0 {
+func (interval *PackageVerInterval) intersects(queryInterval *PackageVerInterval) (intersects bool) {
```

Comment:
Minor consistency suggestion: since we have a comment for `contains`, we could add one for `intersects` as well.